pkg/kafka/auth/config.go

```diff
@@ -85,6 +85,12 @@ func (config *AuthenticationConfig) InitFromViper(configPrefix string, v *viper.
 	}
 
 	config.TLS = tlsClientConfig.InitFromViper(v)
+	if config.TLS.Enabled {
+		config.Authentication = tls
```

Comment:
I don't think this is correct. Other modes can be compatible with TLS according to @jpkrohling, so just having `tls.enabled` should not change the auth scheme.

Reply:
I have removed this and made change to `pkg/kafka/auth/config.go` to load TLS when `tls.enabled=true`